### PR TITLE
Profiles: ENS Expanded State

### PR DESCRIPTION
**Pull request by jxom**
Fixes RNBW-2672

## What changed (plus any additional context for devs)

This PR adds the support for ENS profiles (https://www.figma.com/file/73fCD7saR9UkZy2aXQoQYU/ens-expanded-state?node-id=25%3A10904) on the NFT expanded state.

I also renamed a few of the ENS specific hooks to make more sense:

- `useENSProfile` -> `useENSRegistration` (as it is more registration specific logic)
- `useENSRegistration` -> `useENSSearch`
- `useENSProfileForm` -> `useENSRegistrationForm`

I have also introduced a new `useENSProfile` hook which retrieves ENS profile data from the required sources (combination of The Graph & ENS smart contracts)

## PoW (screenshots / screen recordings)

https://www.loom.com/share/bc4c7ebbfca6484c8d168e2ee5bf9793

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -113,6 +113,7 @@
     "assert": "1.5.0",
     "async-mutex": "0.3.2",
     "asyncstorage-down": "4.2.0",
+    "big-integer": "1.6.51",
```
Comment (jxom) [inline package.json:116]: Needed this package to polyfill `BigInt` for `formatByCoinType` in `@ensdomains/address-encoder`

File: shim.js [hunk 2]
```diff
@@ -8,6 +8,8 @@ import Storage from 'react-native-storage';
 import toLocaleStringPolyfill from '@rainbow-me/helpers/toLocaleStringPolyfill';
 import logger from 'logger';
 
+if (typeof BigInt === 'undefined') global.BigInt = require('big-integer');
```
Comment (jxom) [inline shim.js:11]: Added the polyfill here.

File: src/apollo/queries.ts [hunk 3]
```diff
@@ -242,6 +248,38 @@ export const ENS_REGISTRATIONS = gql`
...
+    registrationDate: number;
+    expiryDate: number;
+    registrant: {
+      id: string;
+    };
+  };
+};
+
+export const ENS_GET_REGISTRATION = gql`
+  query getRegistration($id: ID!) {
```
Comment (jxom) [inline src/apollo/queries.ts:263]: The new queries here I couldn't directly retrieve from the smart contract.

File: src/helpers/ens.ts [hunk 4]
```diff
@@ -507,7 +518,6 @@ const accentColorAtom = atom({
 });
 
 export {
```
Comment (jxom) [inline src/helpers/ens.ts:520]: Feel like we should export the functions directly (it's better for autocompletion). Thoughts?

File: src/hooks/index.js [hunk 5]
```diff
@@ -32,7 +32,13 @@ export { default as useDimensions } from './useDimensions';
 export { default as useDPI } from './useDPI';
 export { default as useEffectDebugger } from './useEffectDebugger';
 export { default as useEmailRainbow } from './useEmailRainbow';
-export { default as useENSProfileForm } from './useENSProfileForm';
+export { default as useENSPendingRegistrations } from './useENSPendingRegistrations';
```
Comment (jxom) [inline src/hooks/index.js:35]: just making it in alphabetical order

File: src/helpers/ens.ts (hunk 4 above)
Comment (estebanmino) [reply src/helpers/ens.ts:520]: yeah I agree

File: src/apollo/queries.ts (hunk 3 above)
Comment (estebanmino) [reply src/apollo/queries.ts:263]: wondering if we should merge this with https://github.com/rainbow-me/rainbow/pull/2997/files/52fe11345822d35661a51b248eb92a2e7d2d6c43#diff-c196b0d95a4c3d59bf1d172d6819e9d4881b07792e69c4d87a20b80d2bd67fdaR241

Comment (estebanmino) [reply src/apollo/queries.ts:263]: export const ENS_REGISTRATIONS = gql`

File: src/components/ens-registration/TextRecordsForm/TextRecordsForm.tsx [hunk 6]
```diff
@@ -8,16 +9,22 @@ import {
   Divider,
   Stack,
 } from '@rainbow-me/design-system';
-import { useENSProfileForm } from '@rainbow-me/hooks';
+import { useENSRegistrationForm } from '@rainbow-me/hooks';
 
-export default function TextRecordsForm() {
+export default function TextRecordsForm({
+  autoFocusKey,
```
Comment (estebanmino) [inline src/components/ens-registration/TextRecordsForm/TextRecordsForm.tsx:15]: image
crash coming from here maybe?

File: src/components/expanded-state/UniqueTokenExpandedState.tsx [hunk 7]
```diff
@@ -138,14 +143,45 @@ const paragraphSpace: Space = '24px';
...
+                borderRadius={24}
+                height={{ custom: 24 }}
+                source={{ uri: titleImageUrl }}
+                width={{ custom: 24 }}
+              />
+            </Bleed>
+          )}
+          {titleEmoji && (
+            <Bleed right="1px">
+              <Heading containsEmoji size={ios ? '23px' : '20px'}>
```
Comment (estebanmino) [inline src/components/expanded-state/UniqueTokenExpandedState.tsx:175]: hm this feels against the design system?

Comment (estebanmino) [reply src/components/expanded-state/UniqueTokenExpandedState.tsx:175]: ah nvm this is not part of design system

File: src/components/expanded-state/UniqueTokenExpandedState.tsx [hunk 8]
```diff
@@ -409,7 +434,13 @@ const UniqueTokenExpandedState = ({
...
                           // @ts-expect-error JavaScript component
                           label={
-                            hasSendButton ? '􀮶 OpenSea' : '􀮶 View on OpenSea'
+                            hasSendButton
+                              ? `􀮶 ${lang.t(
+                                  'expanded_state.unique_expanded.opensea'
+                                )}`
+                              : `􀮶 ${lang.t(
+                                  'expanded_state.unique_expanded.view_on_opensea'
+                                )}`
```
Comment (estebanmino) [inline src/components/expanded-state/UniqueTokenExpandedState.tsx:443]: 🐐

File: src/components/expanded-state/UniqueTokenExpandedState.tsx [hunk 9]
```diff
@@ -444,90 +475,118 @@ const UniqueTokenExpandedState = ({
...
+                        <Section
+                          addonComponent={
+                            hasEditButton && (
+                              <TextButton
+                                align="right"
+                                onPress={handlePressEdit}
+                                size="18px"
+                                weight="bold"
+                              >
+                                Edit
```
Comment (estebanmino) [inline src/components/expanded-state/UniqueTokenExpandedState.tsx:542]: missed opportunity for `lang.t`

File: src/components/expanded-state/ens/InfoRow.tsx [hunk 10]
```diff
@@ -0,0 +1,143 @@
...
+            <Box
+              borderRadius={16}
+              flexShrink={1}
+              onLayout={({
+                nativeEvent: {
+                  layout: { height },
+                },
+              }) => {
+                setIsMultiline(height > 40);
+                setShow(true);
```
Comment (estebanmino) [inline src/components/expanded-state/ens/InfoRow.tsx:106]: maybe moving this method out for easier readability?

File: src/components/expanded-state/ens/InfoRow.tsx [hunk 11]
```diff
@@ -0,0 +1,143 @@
...
+              }}
+              padding={isMultiline ? '15px' : '10px'}
+              style={{
+                backgroundColor: useAccentColor
+                  ? accentColor + '10'
+                  : 'rgba(255, 255, 255, 0.08)',
```
Comment (estebanmino) [inline src/components/expanded-state/ens/InfoRow.tsx:112]: i see this color in the design system as `white80` is that available to use?

File: src/components/expanded-state/unique-token/NFTBriefTokenInfoRow.tsx [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+      </TokenInfoItem>
+      {/* @ts-expect-error JavaScript component */}
+      <TokenInfoItem
+        align="right"
+        color={
+          floorPrice === 'None'
+            ? colors.alpha(colors.whiteLabel, 0.5)
+            : colors.whiteLabel
+        }
+        enableHapticFeedback={floorPrice !== 'None'}
```
Comment (estebanmino) [inline src/components/expanded-state/unique-token/NFTBriefTokenInfoRow.tsx:93]: this was from before this PR but wondering if the string `None` can change in the future

File: src/utils/labelhash.ts [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+export default function labelhash(unnormalisedLabelOrLabelhash: string) {
+  if (unnormalisedLabelOrLabelhash === '[root]') {
+    return '';
+  }
+  return isEncodedLabelhash(unnormalisedLabelOrLabelhash)
+    ? '0x' + decodeLabelhash(unnormalisedLabelOrLabelhash)
+    : keccak256(toUtf8Bytes(normalizeENS(unnormalisedLabelOrLabelhash)));
+}
```
Comment (estebanmino) [inline src/utils/labelhash.ts:51]: same than ens app right?

File: src/components/expanded-state/ens/InfoRow.tsx (hunk 11 above)
Comment (jxom) [reply src/components/expanded-state/ens/InfoRow.tsx:112]: `white80` is 80 opacity, we could add a `white08` variant, however these colors are private in the design system and not exposed as a background color. 
we will need to find a way around this edge case in the design system, but i'll leave that for another PR

File: src/components/expanded-state/ens/InfoRow.tsx (hunk 10 above)
Comment (jxom) [reply src/components/expanded-state/ens/InfoRow.tsx:106]: sounds good!

File: src/utils/labelhash.ts (hunk 13 above)
Comment (jxom) [reply src/utils/labelhash.ts:51]: yup

**Review by estebanmino — APPROVED** [review]
tried android and ios :chef-kiss:

**Pull request merged**